### DISCUSSION TITLE
Data set add to map

### DISCUSSION THREAD

**Discussion by samareshdasglobal** [TerriaJS/terriajs#6683] [Q&A] [answered]
Hello All,
I am new in Terriamap, I want to add a layer to map at initial load, that means data set will be added workbanch during load. I also need to save the layers in session when user add it manually. Now after adding layer if refresh the page layers are disappeared.

**Comment by samareshdasglobal** [comment]
I have fix the problem add the catalogue item to workbench at datacatelog.ts when all data catalogue is available in viewstate

**Comment by AnaBelgun** [comment, marked as answer]
hi @samareshdasglobal it would be great to share your learnings here for other community members. Thank you.
